src/ComputingEnvironment.f90

```diff
@@ -0,0 +1,40 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief Defines the FutilityComputingEnvironment extended type
+!>
+!> This tyep is a simple struct-like object that bundles together commonly used
+!> computing environment-releated objects such as input/output files, log files,
+!> parallel environment objects, and others.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE FutilityComputingEnvironmentModule
+  USE ParallelEnv
+  USE ExceptionHandler
+  USE FileType_Log
+  USE FileType_HDF5
+  USE MemProf
+  USE Times
+  USE FileType_Input
+  USE FileType_XML
+
+  IMPLICIT NONE
+  PRIVATE
+
+  PUBLIC :: FutilityComputingEnvironment
+
+  TYPE :: FutilityComputingEnvironment
```

Comment:
should we call this FutilityComputingEnvironment? or just ComputingEnvironment, as the filename suggests?

Reply:
I added Futility to the names of the module and type since they are the names that are usable in higher level code.  Adding this will prevent possible name collisions, like Brendan suggested in the ticket.

If you really want them consistent, I would change the file name over the module/type name.  I personally don't think it matters, but I'm open to changing it if that's what others want.

Reply:
sure...it seems clunky, but I can understand that benefit.  I would really prefer that the filename matched, but that's really mostly a preference.

Reply:
Alright, I'll go ahead and change the filename when I address other comments.

Reply:
Changed